server/src-lib/Hasura/GraphQL/Execute/Query.hs

```diff
@@ -135,56 +130,49 @@ mkCurPlanTx
   -> HTTP.Manager
   -> [HTTP.Header]
   -> UserInfo
-  -> [G.Directive G.Name]
-  -> FieldPlans
-  -> m (tx EncJSON, GeneratedSqlMap)
-mkCurPlanTx env manager reqHdrs userInfo directives fldPlans = do
+  -> (Q.Query -> Q.Query)
+  -> ExtractProfile
+  -> RootFieldPlan
+  -> (tx EncJSON, Maybe PreparedSql)
+mkCurPlanTx env manager reqHdrs userInfo instrument ep = \case
   -- generate the SQL and prepared vars or the bytestring
-  resolved <- forM fldPlans $ \(alias, fldPlan) -> do
-    fldResp <- case fldPlan of
-      RFPRaw resp                      -> return $ RRRaw resp
-      RFPPostgres (PGPlan q _ prepMap remoteJoins) -> do
-        let args = withUserVars (_uiSession userInfo) $ IntMap.elems prepMap
-        return $ RRSql $ PreparedSql q args remoteJoins
-      RFPActionQuery tx -> pure $ RRActionQuery tx
-    return (alias, fldResp)
-
-  (,) <$> mkLazyRespTx env manager reqHdrs userInfo directives resolved <*> pure (mkGeneratedSqlMap resolved)
+    RFPPostgres ps@(PreparedSql q prepMap remoteJoinsM) ->
+      let args = withUserVars (_uiSession userInfo) prepMap
+          -- TODO this quietly assumes the intmap keys are contiguous
```

Comment:
Does this TODO need to be fixed?

Reply:
No, it's rather a warning for the next person to touch this code. I've made it into a `WARNING` instead.